### ISSUE TITLE
Clean line breaks in configuration metadata descriptions

### ISSUE THREAD

**Issue by YannCebron** [spring-projects/spring-boot#13601]
SB 2.0.3. RELEASE

has

```
    {
      "sourceType": "org.springframework.boot.jta.atomikos.AtomikosProperties$Recovery",
      "defaultValue": 5,
      "name": "spring.jta.atomikos.properties.recovery.max-retries",
      "description": "Number of retry attempts to commit the transaction before throwing an\n exception.",
      "type": "java.lang.Integer"
    }
```

Clients as IDEs shouldn't need to escape/handle such linebreaks.

**Comment by wilkinsona** [comment]
1.5 appears to be affected too. Replacing new lines with `" "` seems reasonable to me.

**Comment by snicoll** [comment]
> Clients as IDEs shouldn't need to escape/handle such linebreaks.

I must be missing something. If you have a very long description and you want to make sure to split it in several lines, you need some way to indicate that. I agree that this deserves a proper note in the documentation but such basic formatting should stay IMO.

**Comment by wilkinsona** [comment]
I consider it to be a bug because I think the formatting of the code in an IDE should be separate from the formatting in an Initializr client. Clients are many and varied so using the formatting that happens to work in an IDE doesn't make sense to me. If a description is so long as to require separate paragraphs, then that should appear in the metadata, but for other line breaks I would expect the client to take care of that using whatever mechanism is appropriate for its UI.

**Comment by snicoll** [comment]
Thanks for the feedback. I agree the formatting is broken that way: the intention of keeping them was to let developers separate a description in separate paragraphs. Perhaps we could keep `\n` if we detect a "new sentence" or something? 

I am a bit nervous to fix this in `1.5.x` without looking at a few examples with existing IDEs.
